src/TypeInference.cpp

```diff
@@ -1900,8 +1924,7 @@ constraintSet TypeInferenceVisitor::visit(Hierarchy_segment *node,
 
 constraintSet TypeInferenceVisitor::visit(Nonblocking_assignment *node,
                                           typeId type) {
-  auto t = freshType();//REG
-  
+  auto t = freshType(); // REG
```

Comment:
Here you need to propagate the wire type and propagate the reg type in the blocking assignment and assignments inside an always block